chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -420,197 +420,206 @@
+                  admin_ip: admin_ip,
+                  provisioner_web: provisioner_web,
+                  web_path: web_path)
+      end
 
-  when /^(hyperv|windows)/ =~ os
-    template "#{tftproot}/adk-tools/build_winpe_#{os}.ps1" do
-      mode 0644
-      owner "root"
-      group "root"
-      source "build_winpe_os.ps1.erb"
-      variables(os: os,
-                admin_ip: admin_ip)
-    end
+    when /^(hyperv|windows)/ =~ os
+      # Windows is x86_64-only
+      os_dir = "#{tftproot}/#{os}"
 
-    directory "#{os_dir}" do
-      mode 0755
-      owner "root"
-      group "root"
-      action :create
-    end
+      template "#{tftproot}/adk-tools/build_winpe_#{os}.ps1" do
+        mode 0644
+        owner "root"
+        group "root"
+        source "build_winpe_os.ps1.erb"
+        variables(os: os,
+                  admin_ip: admin_ip)
+      end
 
-    # Let's stay compatible with the old code and remove the per-version extra directory
-    if File.directory? "#{os_dir}/extra"
-      directory "#{os_dir}/extra" do
-        recursive true
-        action :delete
+      directory "#{os_dir}" do
+        mode 0755
+        owner "root"
+        group "root"
+        action :create
       end
-    end
 
-    link "#{os_dir}/extra" do
-      action :create
-      to "../windows-common/extra"
-    end
+      # Let's stay compatible with the old code and remove the per-version extra directory
+      if File.directory? "#{os_dir}/extra"
+        directory "#{os_dir}/extra" do
+          recursive true
+          action :delete
+        end
+      end
 
-    missing_files = ! File.exists?("#{os_dir}/boot/bootmgr.exe")
-  end
+      link "#{os_dir}/extra" do
+        action :create
+        to "../windows-common/extra"
+      end
 
-  node.set[:provisioner][:available_oses][os] ||= Mash.new
-  if /^(hyperv|windows)/ =~ os
-    node.set[:provisioner][:available_oses][os][:kernel] = "../../../#{os}/#{kernel}"
-    node.set[:provisioner][:available_oses][os][:initrd] = " "
-    node.set[:provisioner][:available_oses][os][:append_line] = " "
-  else
-    node.set[:provisioner][:available_oses][os][:kernel] = "../../../#{os}/install/#{kernel}"
-    node.set[:provisioner][:available_oses][os][:initrd] = "../../../#{os}/install/#{initrd}"
-    node.set[:provisioner][:available_oses][os][:append_line] = append
+      missing_files = !File.exist?("#{os_dir}/boot/bootmgr.exe")
+    end
+
+    node.set[:provisioner][:available_oses][os] ||= Mash.new
+    node.set[:provisioner][:available_oses][os][arch] ||= Mash.new
+    if /^(hyperv|windows)/ =~ os
+      node.set[:provisioner][:available_oses][os][arch][:kernel] = "../../../#{os}/#{kernel}"
+      node.set[:provisioner][:available_oses][os][arch][:initrd] = " "
+      node.set[:provisioner][:available_oses][os][arch][:append_line] = " "
+    else
+      node.set[:provisioner][:available_oses][os][arch][:kernel] = "../../../#{os}/#{arch}/install/#{kernel}"
```

Comment:
Metrics/LineLength: Line is too long. [109/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)